src/tab.rs

```diff
@@ -1394,33 +1394,41 @@ impl Item {
 
         let mut details = widget::column().spacing(space_xxxs);
         details = details.push(widget::text::heading(self.name.clone()));
-        details = details.push(widget::text(format!("Type: {}", self.mime)));
+        details = details.push(widget::text(fl!("type", mime = self.mime.to_string())));
         let mut settings = Vec::new();
-        //TODO: translate!
-        //TODO: correct display of folder size?
         match &self.metadata {
             ItemMetadata::Path { metadata, children } => {
                 if metadata.is_dir() {
-                    details = details.push(widget::text(format!("Items: {}", children)));
+                    details = details.push(widget::text(fl!("items", items = children)));
+                    let size = calculate_dir_size(self.path_opt().unwrap());
```

Comment:
This will run every time the view needs to be remade (after every message) and will block rendering until it is complete. Please comsider using a subscription to load this information in the background

Reply:
sure, I'm not very familiar with subscription in libcosmic, used once. Do you have a working example similar to this case?